docs/yul.rst

```diff
@@ -541,10 +541,22 @@ evaluate to zero values.
 
 In all other situations, expressions have to evaluate to exactly one value.
 
-The ``continue`` and ``break`` statements can only be used inside loop bodies
-and have to be in the same function as the loop (or both have to be at the
-top level). The ``continue`` and ``break`` statements cannot be used
-in other parts of a loop, not even when it is scoped inside a second loop's body.
+A ``continue`` or ``break`` statement can only be used inside the body of a for-loop, as follows.
+Consider the innermost loop that contains the statement.
+The loop and the statement must be in the same function, or both must be at the top level.
+The statement must be in the loop's body block;
+it cannot be in the loop's initialization block or update block.
+It is worth emphasizing that this restriction applies just
+to the innermost loop that contains the ``continue`` or ``break`` statement:
+this innermost loop, and therefore the ``continue`` or ``break`` statement,
+may appear anywhere in an outer loop, possibly in an outer loop's initialization block or update block.
+For example, the following is legal as far as the ``break`` is concerned,
+because it occurs in the body block of the inner loop,
+despite also occurring in the update block of the outer loop:
+
+.. code-block:: yul
+
+    for Init1 Test1 { ... for Init2 Test2 Update2 { ... break ... } ... } Body1
```

Comment:
CI is failing because this does not pass highlighter's syntax checks.

You have 3 options:
- Don't set highlighting language to Yul. It's just pseudo-code after all.
- Force the highlighter to shut up using `:force:`
    ```rst
    .. code-block:: yul
        :force:
    
        for Init1 Test1 { ... for Init2 Test2 Update2 { ... break ... } ... } Body1
    ```
- Use a real code snippet
    ```rst
    .. code-block:: yul

        for {} true { for {} true {} { break } }
        {
        }
    ```

Reply:
Ah, I didn't realize that -- sorry. I like your suggestion to use real Yul code, because I think it makes the example even clearer. Thanks!